src/ex.rs

```diff
@@ -417,13 +417,35 @@ where
     r
 }
 
+/// Checks if `-Z offline` is available if it is it updates the registry and returns true.
+/// If it is not available this returns false.
+fn prep_offline_registry(ex: &Experiment, toolchain: &Toolchain) -> bool {
+    let args = &[
+        "generate-lockfile",
+        "--manifest-path",
+        "Cargo.toml",
+        "-Z",
+        "offline",
+    ];
+    // This nop cargo command is to update the registry
+    // so we don't have to do it for each crate.
+    let args = &["cargo", "search", "lazy_static"];
+    let _ = toolchain.run_cargo(&ex.name, Path::new("."), args, CargoState::Unlocked);
```

Comment:
I would probably do something like `ex.crates().into_iter().find(...)` to get the crate, then call with_work_crate like we do in the main loop.

Reply:
We should assert that this is successful. If it wasn't then the index may not have updated...